testdata/cover/a1_covermode_set.awk

```diff
@@ -0,0 +1,18 @@
+BEGIN {
+  __COVER[1] = 1
```

Comment:
Just thinking about this now: because array keys in AWK are strings, and in GoAWK implemented as `map[string]value`, it will actually be slightly more efficient to have `cover.Annotate` output the keys as strings to begin with, avoiding a number-to-string conversion on each array assignment. Like so:

```awk
__COVER["1"] = 1
...
__COVER["2"] = 1
...
```

Reply:
I don't think this micro-optimization is worth the effort.

Reply:
I don't feel strongly about this one, but note that "the effort" here is trivial, changing `BEGIN { %s[%d]%s }` to `BEGIN { %s["%d"]%s }`. Or in the AST case, changing `NumExpr{Value: float64(n)}` to `StrExpr{Value: strconv.Itoa(n)}`, so it seems to me we might as well.

Reply:
Done.

Reply:
I also opened https://github.com/benhoyt/goawk/issues/157 to add this optimization more generally (after which we could remove the micro-optimization here :-).

Reply:
Btw. Just reallized why I had this unconsious unwillingness to do this optimization. It's because it relies on implementation detail that can change. So it may happen that when it changes, will render this optimization to be de-optimization in fact. Or like you propose, making it useless. Besides, I can imagine the AWK implementation (probably mawk is doing something like this) that is using different tricks to make it fast, even using list underlying representation if the usage pattern for array fits in a usage pattern of list.